deps/build.jl

```diff
@@ -1,68 +1,116 @@
-# Builds GAP with julia GC, compiles/links a few package against it,
-# and exposes the right paths to julia/GAP
+using BinaryProvider
+using Pkg.Artifacts
 
-import Pkg
+# Parse some basic command-line arguments
+const verbose = "--verbose" in ARGS
 
-gap_version = v"4.11.0"
+# Remove deps.jl now, so that we can recognize an aborted build by its absence
+deps_jl = abspath(joinpath(@__DIR__, "deps-$(VERSION).jl"))
+rm(deps_jl, force = true)
 
+
+# TODO: maybe make the artifact loading lazy, for user's who want to use their own GAP???
+# OTOH, these are very few people and I am not sure we should care about them?
+@show artifact_dir = abspath(artifact"gap-4-11-0")
+
+# that artifact directory contains a single subdirectory, with a name like `gap-4.X.Y`;
+# get that name
+content = readdir(artifact_dir)
+@assert length(content) == 1
+gapdirname = content[1]
+
+
+# the location the GAP source code is installed
+gap_src_root = abspath(joinpath(artifact_dir, "$(gapdirname)"))
+
+# the location of the GAP.jl direction, which contains a `pkg` subdir
 extra_gap_root = abspath(joinpath(@__DIR__, ".."))
-gap_root = abspath(joinpath(extra_gap_root, "gap-$(gap_version)"))
-
-println("gap_root = ", gap_root)
-println("extra_gap_root = ", extra_gap_root)
-
-## Find julia binary
-julia_binary = get(ENV, "JULIA_BINARY", Sys.BINDIR)
-
-## Install GAP
-println("Installing GAP ...")
-cd(extra_gap_root)
-run(`rm -rf gap-$(gap_version)`)
-# TODO: gap-system.org availability is flaky, which often breaks CI builds; so use
-# github mirror instead, which is more reliable, albeit somewhat slower
-#filename = download("https://www.gap-system.org/pub/gap/gap-4.$(gap_version.minor)/tar.bz2/gap-$(gap_version).tar.bz2")
-filename = download("https://github.com/gap-system/gap/releases/download/v$(gap_version)/gap-$(gap_version).tar.bz2")
-run(`tar xjf $(filename)`)
-cd("gap-$(gap_version)")
-run(`./configure --with-gc=julia --with-julia=$(julia_binary)`)
-run(`make -j$(Sys.CPU_THREADS)`)
+
+# the location in which we initiate GAP's "out of tree build"
+gap_bin_root = abspath(joinpath(@__DIR__, "build", "$(gapdirname)-julia-$(VERSION)"))
+
+@show gap_src_root
```

Comment:
are these kept to be shown in `--verbose` mode?

Reply:
I followed your good advice to use a single `@info` statement from below.